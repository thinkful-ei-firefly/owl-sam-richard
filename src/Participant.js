import React from 'react';


//props.store -> array of objects
function Participant(props) {
  const peopleIn = props.store.filter(person => person.inSession);
  const peopleOut = props.store.filter(person => !person.inSession);
  const people = peopleIn.concat(peopleOut);
  const peopleList = people.map(function(person) {
    const dot = `dot ${person.inSession ? 'green-dot' : 'grey-dot'}`
    return (
    <div className='person'>
      <img src={person.avatar} alt='avatar' />
      <div>{person.name}</div>
      <div className='status'>
        <span className={dot}></span>
        <div>{person.inSession ? (person.onStage ? 'on stage' : 'in session') : 'left session'}</div>
      </div>
    </div>
    )
  })

  return (
    peopleList
  );
}

export default Participant;