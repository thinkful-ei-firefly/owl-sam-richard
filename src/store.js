const store = [
  {
      id: 1,
      name: 'Koren Templeton',
      avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  },
  {
      id: 2,
      name: 'Caty Flucker',
      avatar:
          'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: false
  },
  {
      id: 3,
      name: 'Axe Kubicka',
      avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
      inSession: false,
      onStage: false
  },
  {
      id: 4,
      name: 'Frank Runciman',
      avatar:
          'https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: false
  },
  {
      id: 5,
      name: 'Ashla Attwool',
      avatar:
          'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  }
];

export default store;