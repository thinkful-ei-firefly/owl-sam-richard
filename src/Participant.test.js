import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Participant from './Participant';
import store from './store';

describe('Participant testing', () => {
  it ('renders without crashing', () => {
    const div= document.createElement('div');
    ReactDOM.render(<Participant store={store} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it ('renders this UI expected', () => {
    const tree=renderer.create(<Participant store={store} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});