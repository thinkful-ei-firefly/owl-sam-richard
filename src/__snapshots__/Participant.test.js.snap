// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant testing renders this UI expected 1`] = `
Array [
  <div
    className="person"
  >
    <img
      alt="avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div>
      Koren Templeton
    </div>
    <div
      className="status"
    >
      <span
        className="dot green-dot"
      />
      <div>
        on stage
      </div>
    </div>
  </div>,
  <div
    className="person"
  >
    <img
      alt="avatar"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <div>
      Caty Flucker
    </div>
    <div
      className="status"
    >
      <span
        className="dot green-dot"
      />
      <div>
        in session
      </div>
    </div>
  </div>,
  <div
    className="person"
  >
    <img
      alt="avatar"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <div>
      Frank Runciman
    </div>
    <div
      className="status"
    >
      <span
        className="dot green-dot"
      />
      <div>
        in session
      </div>
    </div>
  </div>,
  <div
    className="person"
  >
    <img
      alt="avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <div>
      Ashla Attwool
    </div>
    <div
      className="status"
    >
      <span
        className="dot green-dot"
      />
      <div>
        on stage
      </div>
    </div>
  </div>,
  <div
    className="person"
  >
    <img
      alt="avatar"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <div>
      Axe Kubicka
    </div>
    <div
      className="status"
    >
      <span
        className="dot grey-dot"
      />
      <div>
        left session
      </div>
    </div>
  </div>,
]
`;
